### ISSUE TITLE
Update GitFlow documentation

### ISSUE THREAD

**Issue by aaraney** [NOAA-OWP/evaluation_tools#18]
Further elaborate the GitFlow used in this repo. Elucidate on the use of `develop` branch, forking, pull requests, and how to pull upstream changes into your fork to continue work.

**Comment by jarq6c** [comment]
FWIW, here's my current gitflow.

Step 1. Fork the NOAA-OWP/evaluation_tools repository

The instructions located [here](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo) cover forking, cloning a local copy, and updating remotes to keep origin/main synced with upstream/main.

Step 2. Use the [Feature Branch Workflow](https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow)

From origin/main:
```bash
# Create a new branch and switch to it
git checkout -b new-feature

# Make edits using the normal git commands to track changes as necessary
git status
git add path/to/changed/file
git commit -m "My commit message"

# Create a new remote branch and push to origin/new-feature
git push -u origin new-feature
```

Step 3. Submit Pull Request via GitHub.com

Once the new-feature is ready, submit a pull request to merge the new-feature with main on NOAA-OWP/evaluation_tools.

**Comment by aaraney** [comment]
Thanks for the input! That is pretty much my gitflow as well. I set an `upstream` remote using  `git remote add upstream <upstream-remote>`. This lets me switch back to my `main` branch in my local fork and pull changes from the `upstream` remote using `git fetch upstream`. Then I just rebase any changes on top of my local `main` with `git rebase upstream/main`. 

I wrote most of the guide up yesterday and im about to push it up and open up a PR.
